FAO Publications.js

```diff
@@ -98,10 +98,13 @@ function scrape(doc, url) {
 					newItem.abstractNote = child.textContent;
 				}
 			}
-			// DOI: Some docs contain DOI as the last paragraph in abs field
+			// DOI: Some docs contain DOI as a separate paragraph in abs field
+			// use DOI URL as url when DOI exists
 			var DOILead = 'https://doi.org/';
-			if (abs.textContent.includes(DOILead)) {
-				newItem.DOI = abs.textContent.slice(abs.textContent.indexOf(DOILead) + DOILead.length);
+			if (abs.innerText.includes(DOILead)) {
+				var DOIMatch = abs.innerText.match(/https:\/\/doi\.org\/(.+)/i);
+				newItem.DOI = DOIMatch[1];
+				newItem.url = DOIMatch[0];
```

Comment:
I don't think we should use the DOI as the item's URL, for three reasons:

1. `http://www.fao.org/documents/**` URLs (yes, shockingly, no HTTPS whatsoever on a UN site) are the preferred permalinks according to the site.
2. If FAO.org ever goes down, its pages will still be saved in the Wayback Machine. If DOI.org ever goes down, *its* pages, which are just redirects, mostly won't be.
3. We already store the DOI in the DOI field. Also storing it in the URL field won't provide any additional information.

Reply:
I use the DOI link (when possible) for the URL because from 2021 FAO recommends using the DOI link in citation. All FAO publications in 2021, in their "recommended citation" or "required citation" section, use DOI link, e.g.:

* http://www.fao.org/publications/card/en/c/CB3913EN
* http://www.fao.org/publications/card/en/c/CB1841EN
* http://www.fao.org/publications/card/en/c/CB3196EN

The citation text is at the "Cite this content as" section on the web page and the copyright page of the PDF file (click the PDF icon on the right of " Download", usually at the 4th page). For all publications before 2021 (when DOI was not generated for publications) the URL scheme remains unchanged.

By assigning DOI as the item's URL, Zotero can automatically generate the correct citation text when using the "Food and Agriculture Organization of the United Nations" style. This saves a tremendous amount of drudgeries for FAO writers and editors, which was indeed the primary goal I developed this Zotero translator and the corresponding style file.

I fully understand this practice is not in full conformity with the intended usage of Zotero, but I'd really appreciate it if you could let us use it this way.

Reply:
[The style](https://github.com/citation-style-language/styles/blob/2cd6364de42525f99d00b99d790603643ec67a52/food-and-agriculture-organization-of-the-united-nations.csl#L212) as it exists in the CSL styles repo actually handles journal articles correctly already - if an article has both a DOI and a URL, it'll just show the DOI (as a doi.org link). It does not do that for other item types like books. Fixing that should be easy, and it'll let us avoid coupling a translator with a citation style.